### PR TITLE
[Visual] Rename Data

### PR DESCRIPTION
**Pull request by lamriaimen**
According to the guidelines

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp [hunk 1]
```diff
@@ -107,70 +107,107 @@ int VisualModelImplClass = core::RegisterObject("Generic visual model. If a view
...
+    , d_updateNormals   (initData   (&d_updateNormals, true, "updateNormals", "True if normals should be updated at each iteration"))
+    , d_computeTangents (initData   (&d_computeTangents, false, "computeTangents", "True if tangents should be computed at startup"))
+    , d_updateTangents  (initData   (&d_updateTangents, true, "updateTangents", "True if tangents should be updated at each iteration"))
+    , d_handleDynamicTopology (initData   (&d_handleDynamicTopology, true, "handleDynamicTopology", "True if topological changes should be handled"))
+    , d_fixMergedUVSeams (initData   (&d_fixMergedUVSeams, true, "fixMergedUVSeams", "True if UV seams should be handled even when duplicate UVs are merged"))
+    , d_keepLines (initData   (&d_keepLines, false, "keepLines", "keep and draw lines (false by default)"))
+    , d_vertices2       (initData   (&d_vertices2, "vertices", "vertices of the model (only if vertices have multiple normals/texcoords, otherwise positions are used)"))
+    , d_vtexcoords      (initData   (&d_vtexcoords, "texcoords", "coordinates of the texture"))
+    , m_vtexcoords(initData(&d_vtexcoords, "texcoords", "coordinates of the texture"))
```
Comment (alxbilger) [inline Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp]:
```suggestion
```

File: Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp [hunk 2]
```diff
@@ -107,70 +107,107 @@ int VisualModelImplClass = core::RegisterObject("Generic visual model. If a view
...
+    , d_vtangents       (initData   (&d_vtangents, "tangents", "tangents for normal mapping"))
+    , m_vtangents (initData   (&d_vtangents, "tangents", "tangents for normal mapping"))
```
Comment (alxbilger) [inline Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp]:
```suggestion
```

File: Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp [hunk 3]
```diff
@@ -107,70 +107,107 @@ int VisualModelImplClass = core::RegisterObject("Generic visual model. If a view
...
+    , d_vbitangents     (initData   (&d_vbitangents, "bitangents", "tangents for normal mapping"))
+    , m_vbitangents(initData   (&d_vbitangents, "bitangents", "tangents for normal mapping"))
```
Comment (alxbilger) [inline Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp]:
```suggestion
```

File: Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp [hunk 4]
```diff
@@ -107,70 +107,107 @@ int VisualModelImplClass = core::RegisterObject("Generic visual model. If a view
...
+    , d_edges           (initData   (&d_edges, "edges", "edges of the model"))
+    , m_edges(initData   (&d_edges, "edges", "edges of the model"))
```
Comment (alxbilger) [inline Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp]:
```suggestion
```

File: Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp [hunk 5]
```diff
@@ -107,70 +107,107 @@ int VisualModelImplClass = core::RegisterObject("Generic visual model. If a view
...
+    , d_triangles       (initData   (&d_triangles, "triangles", "triangles of the model"))
+    , m_triangles(initData   (&d_triangles, "triangles", "triangles of the model"))
```
Comment (alxbilger) [inline Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp]:
```suggestion
```

File: Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp [hunk 6]
```diff
@@ -107,70 +107,107 @@ int VisualModelImplClass = core::RegisterObject("Generic visual model. If a view
...
+    , d_quads           (initData   (&d_quads, "quads", "quads of the model"))
+    , m_quads  (initData   (&d_quads, "quads", "quads of the model"))
```
Comment (alxbilger) [inline Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp]:
```suggestion
```

File: Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp [hunk 7]
```diff
@@ -107,70 +107,107 @@ int VisualModelImplClass = core::RegisterObject("Generic visual model. If a view
...
     , xformsModified(false)
 {
     m_topology = nullptr;
 
-    addAlias(&fileMesh, "fileMesh");
+    addAlias(&d_fileMesh, "fileMesh");
 
-    m_vertices2     .setGroup("Vector");
+    d_vertices2     .setGroup("Vector");
     m_vnormals      .setGroup("Vector");
```
Comment (alxbilger) [inline Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp:150]: Not d_vnormals?

**Review by alxbilger — CHANGES_REQUESTED** [review]

File: Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp [hunk 8]
```diff
@@ -107,70 +107,101 @@ int VisualModelImplClass = core::RegisterObject("Generic visual model. If a view
...
+    , d_fileMesh          (initData   (&d_fileMesh, "filename", " Path to an ogl model"))
+    , d_texturename       (initData   (&d_texturename, "texturename", "Name of the Texture"))
+    , d_translation     (initData   (&d_translation, Vec3Real(), "translation", "Initial Translation of the object"))
+    , d_rotation        (initData   (&d_rotation, Vec3Real(), "rotation", "Initial Rotation of the object"))
+    , d_scale           (initData   (&d_scale, Vec3Real(1.0, 1.0, 1.0), "scale3d", "Initial Scale of the object"))
+    , d_scaleTex        (initData   (&d_scaleTex, TexCoord(1.f, 1.f), "scaleTex", "Scale of the texture"))
+    , d_translationTex  (initData   (&d_translationTex, TexCoord(0.f, 0.f), "translationTex", "Translation of the texture"))
+    , d_material			(initData	(&d_material, "material", "Material")) // tex(nullptr)
+    , d_putOnlyTexCoords	(initData	(&d_putOnlyTexCoords, (bool) false, "putOnlyTexCoords", "Give Texture Coordinates without the texture binding"))
+    , d_srgbTexturing		(initData	(&d_srgbTexturing, (bool) false, "d_srgbTexturing", "When sRGB rendering is enabled, is the texture in sRGB colorspace?"))
```
Comment (bakpaul) [inline Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp]:
```suggestion
    , d_srgbTexturing		(initData	(&d_srgbTexturing, (bool) false, "srgbTexturing", "When sRGB rendering is enabled, is the texture in sRGB colorspace?"))
```

**Review by bakpaul — CHANGES_REQUESTED** [review]

File: Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp (hunk 7 above)
Comment (lamriaimen) [reply Sofa/Component/Visual/src/sofa/component/visual/VisualModelImpl.cpp:150]: well m_vnormals is a variable that belongs to Sofa/framework/Core/src/sofa/core/visual/VisualState.h

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/Collision/Geometry/src/sofa/component/collision/geometry/TriangleModel.h [hunk 9]
```diff
@@ -151,7 +151,7 @@ class TriangleCollisionModel : public core::CollisionModel
     /** Pointer to the triangle array of this collision model.
      * Will point directly to the topology triangle buffer if only triangles are present. If topology is using/mixing quads and triangles,
      * This pointer will target \sa m_internalTriangles
-     * @brief m_triangles
+     * @brief d_triangles
```
Comment (fredroy) [inline Sofa/Component/Collision/Geometry/src/sofa/component/collision/geometry/TriangleModel.h]:
```suggestion
     * @brief m_triangles
```

File: Sofa/Component/Visual/src/sofa/component/visual/BaseCamera.h [hunk 10]
```diff
@@ -66,26 +66,78 @@ class SOFA_COMPONENT_VISUAL_API BaseCamera : public core::objectmodel::BaseObjec
...
+    Data<double> d_zFar; ///< Camera's zFar
+    Data<bool> d_computeZClip; ///< Compute Z clip planes (Near and Far) according to the bounding box
+    Data<type::Vec3> d_minBBox; ///< minBBox
+    Data<type::Vec3> d_maxBBox; ///< maxBBox
+    Data<unsigned int> d_widthViewport; ///< widthViewport
+    Data<unsigned int> d_heightViewport; ///< heightViewport
+    Data<sofa::helper::OptionsGroup> d_type; ///< Camera Type (0 = Perspective, 1 = Orthographic)
+
+    Data<bool> d_activated; ///< Camera activated ?
+	Data<bool> d_fixedLookAtPoint; ///< keep the lookAt point always fixed
```
Comment (fredroy) [inline Sofa/Component/Visual/src/sofa/component/visual/BaseCamera.h]:
```suggestion
    Data<bool> d_fixedLookAtPoint; ///< keep the lookAt point always fixed
```

**Review by fredroy — APPROVED** [review]

**Review by alxbilger — APPROVED** [review]

**Review by fredroy — APPROVED** [review]

**Pull request merged**